python/tank/platform/engine.py

```diff
@@ -434,6 +498,11 @@ def destroy(self):
         self._invoker = None
         self._async_invoker = None
 
+        # halt metrics dispatching
+        if self._metrics_dispatcher and self._metrics_dispatcher.dispatching:
+            self._metrics_dispatcher.stop()
```

Comment:
minor: for diagnosing issues around shutdown, could be useful to have a debug call just before the stop call as well? Or is that overkill?